File: CS-Examples/04_Sections/AddAndDeleteSections.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace AddAndDeleteSections
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class.
            Document doc = new Document();

            // Load the document from the specified file path.
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\SectionTemplate.docx");

            // Call the AddSection method to add a new section to the document.
            AddSection(doc);

            // Call the DeleteSection method to remove the last section from the document.
            DeleteSection(doc);

            // Specify the output file name.
            string output = "AddAndDeleteSections_out.docx";

            // Save the modified document to the specified file in DOCX 2013 format.
            doc.SaveToFile(output, FileFormat.Docx2013);

            // Dispose of the document object to release resources.
            doc.Dispose();

            // Open the output file using a file viewer.
            FileViewer(output);
        }

        // Method to add a section to the document.
        private void AddSection(Document doc)
        {
            doc.AddSection();
        }

        // Method to delete the last section from the document.
        private void DeleteSection(Document doc)
        {
            doc.Sections.RemoveAt(doc.Sections.Count - 1);
        }
        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/04_Sections/AddAndDeleteSections.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace AddAndDeleteSections
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object.
			Dim doc As New Document()

			' Load a Word document from a specified file path.
			doc.LoadFromFile("..\..\..\..\..\..\Data\SectionTemplate.docx")

			' Call the AddSection subroutine to add a new section to the document.
			AddSection(doc)

			' Call the DeleteSection subroutine to delete the last section of the document.
			DeleteSection(doc)

			' Specify the output file name for the modified document.
			Dim output As String = "AddAndDeleteSections_out.docx"

			' Save the modified document to the specified output file path in Docx2013 format.
			doc.SaveToFile(output, FileFormat.Docx2013)
			' Dispose the document object to release any resources it holds.
			doc.Dispose()

			FileViewer(output)
		End Sub
		' Subroutine to add a new section to the document.
		Private Sub AddSection(ByVal doc As Document)
			doc.AddSection()
		End Sub

		' Subroutine to delete the last section of the document.
		Private Sub DeleteSection(ByVal doc As Document)
			doc.Sections.RemoveAt(doc.Sections.Count - 1)
		End Sub
		
		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
